app/src/main/java/com/spoony/spoony/presentation/report/ReportRoute.kt

```diff
@@ -52,22 +50,26 @@ fun ReportRoute(
     val lifecycleOwner = LocalLifecycleOwner.current
 
     val state by viewModel.state.collectAsStateWithLifecycle(lifecycleOwner = lifecycleOwner)
+
     ReportScreen(
-        options = state.options,
-        selectedOption = state.selectedOption,
-        reportContext = state.reportContext
+        reportOptions = state.reportOptions,
+        selectedReportOption = state.selectedReportOption,
+        reportContext = state.reportContext,
+        onOptionReportSelected = { viewModel.updateSelectedReportOption(it) },
+        onContextChanged = { viewModel.updateReportContext(it) }
```

Comment:
p3) reflection에 대해서 알고계신가요? 장단점을 찾아보시고 적용하시면 좋을 것 같습니다.

```suggestion
        onOptionReportSelected = viewModel::updateSelectedReportOption,
        onContextChanged = viewModel::updateReportContext
```

Reply:
### 비교 요약

특징 | 람다 사용 | 메서드 참조 사용
-- | -- | --
코드 | { viewModel.updateData(it) } | viewModel::updateData
가독성 | 명확한 흐름 제공, 특히 추가 로직이 필요한 경우 적합 | 코드가 간결하고 깔끔함
유연성 | 추가 연산, 조건문, 데이터 변환 등 다양한 작업 가능 | 메서드 호출 외의 작업은 불가능
성능 | 약간의 오버헤드 (보통 무시할 수 있음) | 더 효율적 (람다 객체 생성 없음)
중복 가능성 | 단순 메서드 호출의 경우 불필요한 코드 중복 발생 가능 | 코드 중복을 방지할 수 있음
확장성 | 추가 작업이 필요한 복잡한 로직에 적합 | 단순한 메서드 호출에 적합

정도 있는 것 같습니다. 덕분에 알아갑니다 감사합니다..! 수정하겠습니다